.github/workflows/jest-test.yaml

```diff
@@ -0,0 +1,44 @@
+name: Run Jest Tests
+
+on:
+  push:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v2
+
+    - name: Setup Node.js
+      uses: actions/setup-node@v2
+      with:
+        node-version: '18'
+
+    - name: Cache node modules
+      uses: actions/cache@v2
+      with:
+        path: app/node_modules
+        key: ${{ runner.OS }}-node-${{ hashFiles('app/package-lock.json') }}
```

Comment:
We're using yarn for this project, do we have a working equivalent with yarn? Same for below